native_client/deepspeech.cc

```diff
@@ -20,23 +21,35 @@
 #include "ctcdecode/ctc_beam_search_decoder.h"
 
 //TODO: infer batch size from model/use dynamic batch size
-const unsigned int BATCH_SIZE = 1;
+constexpr unsigned int BATCH_SIZE = 1;
 
 //TODO: use dynamic sample rate
-const unsigned int SAMPLE_RATE = 16000;
+constexpr unsigned int SAMPLE_RATE = 16000;
 
-const float AUDIO_WIN_LEN = 0.025f;
-const float AUDIO_WIN_STEP = 0.01f;
-const unsigned int AUDIO_WIN_LEN_SAMPLES = (unsigned int)(AUDIO_WIN_LEN * SAMPLE_RATE);
-const unsigned int AUDIO_WIN_STEP_SAMPLES = (unsigned int)(AUDIO_WIN_STEP * SAMPLE_RATE);
+constexpr float AUDIO_WIN_LEN = 0.032f;
+constexpr float AUDIO_WIN_STEP = 0.02f;
+constexpr unsigned int AUDIO_WIN_LEN_SAMPLES = (unsigned int)(AUDIO_WIN_LEN * SAMPLE_RATE);
+constexpr unsigned int AUDIO_WIN_STEP_SAMPLES = (unsigned int)(AUDIO_WIN_STEP * SAMPLE_RATE);
 
-const unsigned int MFCC_FEATURES = 26;
+constexpr unsigned int MFCC_FEATURES = 26;
 
-const float PREEMPHASIS_COEFF = 0.97f;
-const unsigned int N_FFT = 512;
-const unsigned int N_FILTERS = 26;
-const unsigned int LOWFREQ = 0;
-const unsigned int CEP_LIFTER = 22;
+constexpr float PREEMPHASIS_COEFF = 0.97f;
+constexpr unsigned int N_FFT = 512;
+constexpr unsigned int N_FILTERS = 26;
+constexpr unsigned int LOWFREQ = 0;
+constexpr unsigned int CEP_LIFTER = 22;
+
+constexpr size_t WINDOW_SIZE = AUDIO_WIN_LEN * SAMPLE_RATE;
+
+std::array<float, WINDOW_SIZE> calc_hamming_window() {
```

Comment:
This function (and then `hamming_window`) could be made constexpr too with C++14 and some effort (like a constexpr `cos` function), but I'm leaving that as a future optimization for when C++14 stdlibs are more commonly available.